src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -271,10 +271,17 @@ export default class Fms {
 
         this._verifyRouteContainsMultipleWaypoints();
         this._initializeFlightPhaseForCategory(category);
-        this._initializeDepartureAirport(origin);
-        this._initializeDepartureRunway();
-        this._initializeArrivalAirport(destination);
-        this._initializeArrivalRunway();
+
+        if (category === FLIGHT_CATEGORY.DEPARTURE) {
+            this._initializeDepartureAirport(origin);
+            this._initializeDepartureRunway();
+        }
+
+        if (category === FLIGHT_CATEGORY.ARRIVAL) {
+            this._initializeArrivalAirport(destination);
+            this._initializeArrivalRunway();
+        }
```

Comment:
What if they depart and arrive at fields which are both under our control? This will be possible.

Reply:
I should add part of the original intent of the `origin` / `destination` fields was that they were intended to replace the `category` (since they are rather duplicitave). I can't remember the reason why we didn't just do it at that time though, so maybe there's something it would have broken.

But the `category` alone will not support all the required combinations, hence moving to just defining if and where they should (a) spawn on the ground at, and/or (b) which airport they are landing at.

Reply:
If as per other discussion in #1762 there is a desire to have both the origin and destination sections in the flight strip be populated (for immersive reasons) then you once again *cannot* rely purely on presence/absence of `origin` and `destination` properties in the `aircraftInitProps` passed to `Fms` constructor, because both will be populated with values to be shown on the flight strip -- [here](https://github.com/openscope/openscope/blob/c391f605530dac4c5e9f8554d4ecf56c5b17f28d/src/assets/scripts/client/aircraft/AircraftModel.js#L506) is the call from `AircraftModel` constructor which passes along the same set of initialization properties that it received from `AircraftController._createAircraftWithInitializationProps()`. `AircraftModel` will itself also [parse](https://github.com/openscope/openscope/blob/c391f605530dac4c5e9f8554d4ecf56c5b17f28d/src/assets/scripts/client/aircraft/AircraftModel.js#L521) the same set of init props to obtain the [values](https://github.com/openscope/openscope/blob/c391f605530dac4c5e9f8554d4ecf56c5b17f28d/src/assets/scripts/client/aircraft/AircraftModel.js#L685-L686) that ultimately show up in the flight strip.

Even overflights could/would have `origin` and `destination` values.

> `category` alone will not support all the required combinations

Then perhaps we will need to consider expanding `category` to support four possibilities: "arriving here", "departing here", "neither (overflight)" and "both". (where "here" currently means one airport but in future could be a TRACON with multiple fields)

Reply:
I can't imagine a scenario where the `category` is a necessary parameter; everything can be more clearly inferred from the origin and destination. The mere existence of `category` could only serve to confusion when inevitably someone specifies a "departure" category with no origin, etc. Since we have to specify _where_ they're departing/arriving already, there's no reason that they _are_ a departure/arrival.

The conversation in #1762 is agreed to this effect, with the change being not to use a `origin`/`destination` property on the spawnPattern, but instead to specify origin/destination in the airline list for a given spawnPattern.

> Even overflights could/would have origin and destination values.

This is the desired behavior. All aircraft in the simulation would have origin/destination values (unless they weren't specified, for example with an overflight). We _do not need_ to know an aircraft's "category" for anything the sim does. There are places that we have logic branches based on its value, but these can and should be handled in a different way anyway.

For example, in the current implementation, there is only one airport that we control at a time. An aircraft is an arrival if `aircraftModel.fms.arrivalAirportModel.icao === AirportController.current.icao`. If an aircraft is being spawned and `aircraftModel.fms.departureAirportModel.icao === AirportController.current.icao`, then we know to spawn it on the ground, etc.

The issue is that what we truly cannot do is add support for multiple airports as long as decisions are made based on a "category". Every aircraft is a departure, overflight, _and_ arrival-- we shouldn't be labeling the aircraft as one thing for its whole life, as it is only a matter of the perspective of the where the controller is along their route. That's why we want to move to specifying where they start and end, and make all our decisions based on that alone.

Reply:
Removed `category` check and instead rely purely on whether `origin`/`destination` refer to current airport to decide whether or not to initialize the `departureAirportModel`+`departureRunwayModel` / `arrivalAirportModel`+`arrivalRunwayModel` respectively.

I'm not thoroughly convinced this is fully safe -- what if both origin and destination are set as current airport, would that surface other problems elsewhere, etc.

But it does at least still guarantee that [the reported null value](https://github.com/openscope/openscope/issues/1762#issuecomment-778681639) is eliminated so I'll leave it at that.

Reply:
> I'm not thoroughly convinced this is fully safe -- what if both origin and destination are set as current airport, would that surface other problems elsewhere, etc.

Right, a valid concern, and it might. My point is that we have to face those issues anyway in the effort to move to supporting various future functionality, so avoiding them is just further delaying the transition. 👍